core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.html

```diff
@@ -1,4 +1,25 @@
 <div class="oppia-contributions-show-review">
+  <div class="oppia-contributions-dropdown-container">
+    <label for="oppia-contributions-tasklist-dropdown">View: </label>
+    <select name="oppia-contribution-tasks"
+            id="oppia-contributions-tasklist-dropdown"
+            ng-model="$ctrl.activeDropdownTabChoice"
+            ng-change="$ctrl.switchToTab($ctrl.activeDropdownTabChoice.split(' ')[0], $ctrl.activeDropdownTabChoice.split(' ')[1])">
```

Comment:
I think there's a bit too much logic in the template. Perhaps better to have that be in the TS and have functions such as getActiveDropdownTabType(), etc.

Ditto for getting the values for the value arg. Try to encapsulate the logic in unit-testable functions; the HTML templates should be as declarative and straightforward as possible.

Reply:
@seanlip w.r.t. the first half of your comment, Instead of directly calling `switchToTab()` when the dropdown selection changes, I've added a function(`switchToTabFromDropdownChoice()`) for that purpose (which itself then proceeds to call `switchToTab()` with the appropriate arguments).

As for the second half of your comment (regarding the value attribute), the simplest way I could think of to set the dropdown options' value attributes, was to add an additional property (`dropdownOptionValue`) to each of the objects in the existing `contributionTabs` and `reviewTabs` arrays, and then directly accessing them in the template. I've made changes for the same.

Finally, I've added a unit test for the new function. Please take a look at the same.

Reply:
I left another note -- thanks!